simulator/modules/garbage.py

```diff
@@ -1,96 +1,102 @@
-import simpy
+from utils import get_logger
 
 class GC(object):
 
-    def __init__(self, env, server, threshold=0.9, sleep_time=0.02):
+    def __init__(self, env, server, conf, log_path):
         self.env = env
         self.server = server
         self.heap = server.heap
-        self.threshold = threshold
-        self.sleep_time = sleep_time
-        self.collecting_trash = False
+        self.threshold = float(conf['threshold'])
+        self.sleep_time = float(conf['sleep_time'])
+
+        self.is_gcing = False
         self.times_performed = 0
         self.collects_performed = 0
         self.gc_exec_time_sum = 0
         self.gc_process = self.env.process(self.run())
 
+        self.logger = get_logger(log_path + "/gc.log", "GC")
+
     def run(self):
         while True:
-            if self.heap.level >= self.threshold and not self.collecting_trash:
-                gc_start_time = self.env.now
+            if self.heap.level >= self.threshold and not self.is_gcing:
                 yield self.env.process(self.collect())
-                self.gc_exec_time_sum += (self.env.now - gc_start_time)
                 self.times_performed += 1
 
             yield self.env.timeout(self.sleep_time)  # wait for...
 
     def collect(self):
-        print("At %.3f, GC is running. We have %.3f of trash" % (self.env.now, self.heap.level))
-        self.collecting_trash = True
+        self.logger.info(" At %.3f, GC is running. We have %.3f of trash" % (self.env.now, self.heap.level))
+        self.is_gcing = True
         self.server.action.interrupt()
 
+        gc_start_time = self.env.now
         while self.heap.level > 0:                                          # while threshold is empty...
             trash = self.heap.level                                         # keeps the amount of trash
             yield self.env.timeout(self.gc_execution_time_by_trash(trash))  # run the time of discarting
             yield self.heap.get(trash)                                      # discards the trash
+        self.gc_exec_time_sum += (self.env.now - gc_start_time)
 
-        self.collecting_trash = False
-        print("At %.3f, GC finish his job. Now we have %.3f of trash" % (self.env.now, self.heap.level))
+        self.is_gcing = False
+        self.logger.info(" At %.3f, GC finish his job. Now we have %.3f of trash\n" % (self.env.now, self.heap.level))
         self.server.action = self.env.process(self.server.run())
         self.collects_performed += 1
 
     def gc_execution_time_by_trash(self, trash):
         # TODO(David) implement the way to calculate the execution time of Garbage Collector
         return trash
 
-
 class GCI(object):
 
-    def __init__(self, env, server,  threshold=0.7, check_heap=2, initial_estimated_gc_exec_time=0.0, sleep_time=0.00002):
+    def __init__(self, env, server,  conf, log_path):
```

Comment:
I am almost convinced that you guys should go through the process of implementing the real GCI in python (move https://github.com/danielfireman/gci/tree/master/python and move to https://github.com/gcinterceptor and update).

This simulator entity could use the same source as any other program. What do you guys think?

Reply:
We have talked with @thiagomanel and we decided to think about it later, to evaluate the situation better.

Reply:
Ack.